gnc/sub8_trajectory_generator/src/sub8_state_space.cc

```diff
@@ -24,13 +24,13 @@ double Sub8StateSpace::distance(const State* state1,
   // return LQR cost to go to state 2 from state 1? 
 
   // naive implementation is euclidean distance in position
-  std::vector<double> pos1;
-  std::vector<double> pos2;
+  Vector3d pos1; 
+  Vector3d pos2; 
   state1->as<Sub8StateSpace::StateType>()->getPosition(pos1);
   state2->as<Sub8StateSpace::StateType>()->getPosition(pos2);
 
-  return sqrt(pow(pos2[0] - pos1[0], 2) +
-              pow(pos2[1] - pos1[1], 2 + pow(pos2[2] - pos1[2], 2)));
+  return sqrt(pow(pos2(0) - pos1(0), 2) +
```

Comment:
Is it possible to use Eigen here? Or are you trying to avoid casting overhead?

*.norm()?